psydac/linalg/stencil.py

```diff
@@ -894,6 +916,59 @@ def _tocoo_no_pads( self ):
 
         return M
 
+    def tocoo_local( self ):
```

Comment:
Why do you need this function? By calling `tosparse(with_pads=False)` you will get exactly the same non-zero elements in the output sparse matrix, the only difference being that the indices will be global instead of local.

Reply:
yes I need to local indices here

Reply:
I see. This function is almost identical to `tosparse`, and it would be nice to avoid such a large code duplication.

Is there not a way to have the `tosparse` function return what you need, maybe by passing the flag `local=True`?